agreement_maker/make_agreement.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python3
+import os
+import sys
+import argparse
+import logging
+from typing import Tuple
+import numpy as np
+import rasterio
+import rioxarray as rxr
+import xarray as xr
+import gval
+from dask.distributed import Client, LocalCluster
+from dask import delayed
+from pythonjsonlogger import jsonlogger
+from osgeo import gdal
+
+# -----------------------------------------------------------------------------
+# GLOBAL GDAL CONFIGURATION
+# -----------------------------------------------------------------------------
+gdal.SetConfigOption("GDAL_CACHEMAX", os.getenv("GDAL_CACHEMAX"))
```

Comment:
Is there a need to set the config in Python and not directly via environment variables? Doing through env variables is a superior strategy and give flexibility. If there is no specific need I would suggest we control such config through env variables across all jobs for consistency.